engine/src/org/pentaho/di/cluster/SlaveServer.java

```diff
@@ -266,13 +267,13 @@ public String getXML() {
     return xml.toString();
   }
 
-  public Object clone() {
+  public synchronized Object clone() {
```

Comment:
Same for here.
Lets assume clone is not synchronized here.
Lets assume that Thread-1 is currently executeing replaceMeta() and it changed half of the fields of sharedSlaveServer.

Thread-2 calls clone method and clone method passes this reference to replace method. But this is not consistent, becaue not all properties were replaced yet. So, we get half of the properties replaced, half - not yet. Thats why synchronization is needed either.

Agree?